tap_gorgias/streams.py

```diff
@@ -53,189 +117,142 @@ class Tickets(Stream):
-            for record in data[self.results_key]:
-                yield record
-            url = data['meta']['next_items']
-            LOGGER.info(f'next url for GET: {url}')
-            if not url:
-                break
+    # There are two APIs that return the same data:
+    # 1. the views API, https://developers.gorgias.com/reference/get_api-views
+    # 2. the tickets API, https://developers.gorgias.com/reference/get_api-tickets
 
     def sync(self, state, config):
+        # https://developers.gorgias.com/reference/get_api-tickets
         view_id = config.get(self.view_id_key)
         if not view_id:
+            # This API doesn't require the view ID, but to preserve previous behaviour,
+            # exit when this not provided in the config
             LOGGER.exception(f'No view ID provided for {self.name}')
             return
 
-        try:
-            sync_thru = singer.get_bookmark(state, self.name, self.replication_key)
-        except TypeError:
-            sync_thru = self.start_date
-
-        max_synced_thru = max(sync_thru, self.start_date)
-
-        messages_stream = Messages(self.client)
-
-        url = self.url.format(view_id)
-
-        LOGGER.info(f'starting url for Paging GET: {url}')
-
-        # tickets are retrieved in descending order based on updated_datetime
-        # with no date filtering
-        for row in self.paging_get(url):
-            ticket = {k: self.transform_value(k, v) for (k, v) in row.items()}
-            curr_synced_thru = ticket[self.replication_key]
+        sync_thru, max_synced_thru = self.get_sync_thru_dates(state)
+        # Since there are no datetime filters available for this endpoint,
+        # sort in descending order and stop when we've reached the bookmark
+        query_params = {
+            'view_id': view_id,
+            'limit': 100,
+            'order_by': 'created_datetime:desc',
+        }
+        LOGGER.info(f'Starting fetch for {self.name} stopping at {sync_thru}')
+        for row in self.cursor_get(self.url, query_params):
+            message = {k: self.transform_value(k, v) for (k, v) in row.items()}
+            curr_synced_thru: str = message[self.replication_key]
+            max_synced_thru = max(curr_synced_thru, max_synced_thru)
             if curr_synced_thru > sync_thru:
-                yield(self.stream, ticket)
-                max_synced_thru = max(curr_synced_thru, max_synced_thru)
-                if messages_stream.is_selected():
-                    yield from messages_stream.sync(ticket['id'], sync_thru)
+                yield(self.stream, message)
             else:
                 break
 
         self.update_bookmark(state, max_synced_thru)
 
 
-class Messages(Stream):
+class Messages(CursorStream):
     name = 'messages'
     replication_method = 'INCREMENTAL'
     key_properties = ['id']
-    replication_key = 'sent_datetime'
-    url = '/api/tickets/{}/messages'
+    replication_key = 'created_datetime'
```

Comment:
Changed the replication key since `sent_datetime` can be None and it'll fail the comparison